packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

```diff
@@ -17,81 +19,263 @@ import {
+                            return
+                        }
+
+                        const newData = clone(oldData)
+
+                        const listIndex = newData.data.findIndex(({id}) => id === parameters.listId)
+
+                        // Return undefined if no list is found
+                        if (listIndex < 0) {
+                            return
+                        }
+
+                        // Remove the list from the result object
+                        newData.data.splice(listIndex, 1)
+                        newData.limit--
+                        newData.total--
+
+                        return newData
+                    }
+                }
+            ],
             remove: [{queryKey: getCustomerProductList.path(parameters)}]
         }
     },
     deleteCustomerProductListItem(customerId, {parameters}) {
         return {
-            // TODO: Rather than invalidate, can we selectively update?
-            invalidate: [
-                {queryKey: getCustomerProductList.queryKey(parameters)},
-                {queryKey: getCustomerProductLists.queryKey(parameters)}
+            update: [
+                {
+                    queryKey: getCustomerProductList.queryKey(parameters),
+                    updater: (oldData: CustomerProductList) => {
+                        if (!oldData) {
+                            return
+                        }
+
+                        const newData = clone(oldData)
+
+                        const itemIndex = newData.customerProductListItems?.findIndex(
+                            ({id}) => id === parameters.itemId
+                        )
+
+                        // Return undefined if there is no item found.
+                        if (typeof itemIndex === 'undefined' || itemIndex < 0) {
+                            return
+                        }
+
+                        // Remove the list item
+                        newData.customerProductListItems?.splice(itemIndex, 1)
+
+                        return newData
+                    }
+                },
+                {
+                    queryKey: getCustomerProductLists.queryKey(parameters),
+                    updater: (oldData: CustomerProductListResult) => {
+                        if (!oldData) {
+                            return
+                        }
+
+                        const newData = clone(oldData)
+
+                        const listIndex = newData?.data.findIndex(
+                            ({id}) => id === parameters.listId
+                        )
+                        const itemIndex = newData?.data?.[
+                            listIndex
+                        ]?.customerProductListItems?.findIndex(({id}) => id === parameters.itemId)
+
+                        // Return undefined if no item was found in the provided list.
+                        if (listIndex < 0 || typeof itemIndex === 'undefined' || itemIndex < 0) {
```

Comment:
```suggestion
                        if (typeof itemIndex === 'undefined' || itemIndex < 0) {
```
`itemIndex` will always be undefined if `listIndex < 0`, so you don't need to check both.

Reply:
Thats true. 👍